### PR TITLE
Iframe from templated src

### PR DESCRIPTION
**Pull request by alecsammon**
Allow constructing an iframe with a string with multiple replacements.

Current the iframe plugin allows adding a source with a static string, or a reference to an annotation.

This extends the plugin to allow creating a source from a pattern and multiple annotations.
This is useful if you want to point at another tool and include the entity name as a query parameter.
Or link to documentation depending on the language of the component.

#### :heavy_check_mark: Checklist

- [x] Added tests for new functionality and regression tests for bug fixes
- [x] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [x] Added or updated documentation (if applicable)

### CODE REVIEW

File: plugins/frontend/backstage-plugin-iframe/src/components/IFrameComponent.tsx [hunk 1]
```diff
@@ -109,7 +152,14 @@ const IFrameFromSrc = (props: IFrameProps) => {
  * @public
  */
 export const IFrameCard = (props: IFrameComponentProps) => {
-  const { src, srcFromAnnotation, height, width, title } = props;
+  const {
+    src,
+    srcFromAnnotation,
+    srcWithAnnotationReplacements,
```
Comment (Xantier) [inline plugins/frontend/backstage-plugin-iframe/src/components/IFrameComponent.tsx:158]: Let's call this something like templatedSrc or srcWithTemplate or similar I think

**Review by Xantier — COMMENTED** [review]
A naming thing and tests look to fail but other than those, this is nearly good to go

File: plugins/frontend/backstage-plugin-iframe/src/components/IFrameComponent.tsx (hunk 1 above)
Comment (alecsammon) [reply plugins/frontend/backstage-plugin-iframe/src/components/IFrameComponent.tsx]: Agree! Thanks.

Updated to `templatedSrc`

**Comment by alecsammon** [comment]
Thanks @Xantier  - I've update the naming as suggested, and fixed a problem with the version of `@backstage/catalog-model`. 

Build is now 🟢 

When you have time to review again that would be greatly appreciated.

**Review by Xantier — APPROVED** [review]
Nice!

**Comment by alecsammon** [comment]
Thank you ❤️

**Pull request merged**